shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/data/repository/FeedingPointRepositoryImpl.kt

```diff
@@ -1,57 +1,84 @@
+import com.epmedu.animeal.feeding.data.model.CategoryTagData.Cats
+import com.epmedu.animeal.feeding.data.model.FeedingData
+import com.epmedu.animeal.feeding.data.model.FeedingPointData
+import com.epmedu.animeal.feeding.data.model.FeedingPointDataStatus.Fed
 import com.epmedu.animeal.feeding.domain.model.Feeder
 import com.epmedu.animeal.feeding.domain.model.FeedingPoint
-import com.epmedu.animeal.feeding.domain.model.enum.AnimalPriority
 import com.epmedu.animeal.feeding.domain.model.enum.AnimalState
 import com.epmedu.animeal.feeding.domain.repository.FeedingPointRepository
 import com.epmedu.animeal.feeding.presentation.model.MapLocation
 import com.epmedu.animeal.foundation.switch.AnimalType
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.flowOf
-import kotlin.random.Random
-import kotlin.random.nextInt
-
-class FeedingPointRepositoryImpl : FeedingPointRepository {
-
-    private val stubData: List<FeedingPoint> = List(25) { index ->
-        FeedingPoint(
-            index,
-            title = "$index - Near to Bukia Garden M.S Technical University",
-            "$index - This area covers about 100 sq.m. -S, it starts with Bukia Garden " +
-                "and Sports At the palace. There are about 1000 homeless people here " +
-                "The dog lives with the habit of helping You need.",
-            AnimalPriority.values().random(),
-            AnimalState.values().random(),
-            AnimalType.values().random(),
-            Random.nextBoolean(),
-            location = MapLocation(
-                Random.nextDouble(41.6752, 41.7183),
-                Random.nextDouble(44.7724, 44.8658)
-            ),
-            lastFeeder = Feeder(
-                id = index,
-                name = "$index - Giorgi Abutidze",
-                time = "${Random.nextInt(0..24)} hours ago"
-            )
-        )
-    }
 
-    override fun getAllFeedingPoints(): Flow<List<FeedingPoint>> = flowOf(stubData)
+internal class FeedingPointRepositoryImpl(
+    private val authAPI: AuthAPI,
+    private val favouriteApi: FavouriteApi,
+    private val feedingPointApi: FeedingPointApi
+) : FeedingPointRepository {
+
+    private var feedingPoints: List<FeedingPoint> = emptyList()
+
+    override fun getAllFeedingPoints(): Flow<List<FeedingPoint>> {
+        return feedingPointApi.getAllFeedingPoints()
+            .combine(favouriteApi.getFavouriteList(authAPI.currentUserId)) { feedingPointDataList, favouriteDataList ->
+                feedingPointDataList.map { feedingPointData ->
+                    feedingPointData.toFeedingPoint(
+                        isFavourite = favouriteDataList.any { it.feedingPointId == feedingPointData.id }
+                    )
+                }.also {
+                    feedingPoints = it
+                }
+            }
+    }
 
     override fun getCats(): Flow<List<FeedingPoint>> = flowOf(
```

Comment:
Seems like all the methods that rely on `feedingPoints` won't work if you don't call `getAllFeedingPoints` first.
Maybe, we can call `getAllFeedingPoints` if `feedingPoints` is empty

Reply:
Good point. Replaced list with flow and lazy block